src/client/views/cube-view/cube-view.tsx

```diff
@@ -407,27 +428,60 @@ export class CubeView extends React.Component<CubeViewProps, CubeViewState> {
   }
 
   getStoredLayout(): CubeViewLayout {
-    return localStorage.get("cube-view-layout") || { dimensionPanelWidth: 240, pinboardWidth: 240 };
+    return localStorage.get("cube-view-layout-ver2") || defaultLayout;
   }
 
   storeLayout(layout: CubeViewLayout) {
-    localStorage.set("cube-view-layout", layout);
+    localStorage.set("cube-view-layout-ver2", layout);
   }
 
-  onDimensionPanelResize(value: number) {
-    let { layout } = this.state;
-    layout.dimensionPanelWidth = value;
-
+  private updateLayout(layout: CubeViewLayout) {
     this.setState({ layout });
     this.storeLayout(layout);
   }
 
-  onPinboardPanelResize(value: number) {
-    let { layout } = this.state;
-    layout.pinboardWidth = value;
+  toggleDimensionPanel() {
+    const { layout: { dimensionPanel }, layout } = this.state;
+    this.updateLayout({
+      ...layout,
+      dimensionPanel: {
+        ...dimensionPanel,
+        hidden: !dimensionPanel.hidden
+      }
+    });
+  }
 
-    this.setState({ layout });
-    this.storeLayout(layout);
+  togglePinboard() {
```

Comment:
a bit of a duplication down here 3 times :)

Reply:
Even four times. I need some solution for deep updates in objects but didn't have courage to pick new dependency for that.

Reply:
Will be resolved in #202